tests/integration/test_chat.py

```diff
@@ -386,6 +387,39 @@ def test_button_disabled_after_click(self):
         for button in buttons:
             self.assertTrue(button.get_attribute('disabled'))
 
+    @patch('chat.chat.BUTTONS_LEAVING_TRANSITION_DURATION', 3000)
+    def test_button_disable_handler_after_click(self):
+        self.load_scenario('xml/chat_defaults.xml')
+        self.wait_until_buttons_are_displayed()
+        chat_block = self.element.find_element_by_class_name('chat-block')
+        buttons = self.element.find_elements_by_css_selector('.buttons .response-button')
+        # Patch `_is_final_step` instead of `submit_response` since `submit_response` needs to return a
+        # response or other errors might be raised. `_is_final_step` is a good proxy since it's
+        # called each time `submit_response` is called.
+        with patch('chat.chat.ChatXBlock._is_final_step') as mocked_method:
+            self.browser.execute_script("arguments[0].click();", buttons[0])
+            time.sleep(0.25)
```

Comment:
We try to never use explicit `time.sleep` in tests (edX won't accept PRs with `time.sleep` in tests). This seems to work for me:

```python
@patch('chat.chat.BUTTONS_LEAVING_TRANSITION_DURATION', 50)
def test_button_disable_handler_after_click(self):
    self.load_scenario('xml/chat_defaults.xml')
    self.wait_until_buttons_are_displayed()
    chat_block = self.element.find_element_by_class_name('chat-block')
    buttons = self.element.find_elements_by_css_selector('.buttons .response-button')
    # Patch `_is_final_step` instead of `submit_response` since `submit_response` needs to return a
    # response or other errors might be raised. `_is_final_step` is a good proxy since it's
    # called each time `submit_response` is called.
    with patch('chat.chat.ChatXBlock._is_final_step') as mocked_method:
        self.browser.execute_script("arguments[0].click();", buttons[0])
        self.browser.execute_script("arguments[0].click();", buttons[1])
        self.browser.execute_script("arguments[0].click();", buttons[0])
        self.wait_until_buttons_are_displayed()
        self.assertEquals(mocked_method.call_count, 1)
```

I used `50` for the buttons leaving duration animation because with lower values, selenium does not seem to be able to click the buttons three times quickly enough before they get removed from the DOM.
I was able to remove `time.sleep` calls by adding `self.wait_until_buttons_are_displayed()` after the clicks - the reason why this works is that that method waits for the next set of buttons (in the `entering` state) to be displayed, and because of the way the JS code is currently written, the new buttons will only enter *after* the XHR requests are completed.

Reply:
OK! Thanks for all the help!